utils.py

```diff
@@ -40,12 +42,35 @@
 """
 def csv_helper(fpath, headers):
     result = []
-    with open(fpath) as f:
+    with codecs.open(fpath, encoding='utf-8') as f:
         for line in f.readlines()[1:]:
             csv_data = line.strip().split(",")
             result.append(dict(zip(headers,csv_data)))
     return result
 
+def filter_entity(entity_list, search_fields, keywords):
+    recall_set = []
+    for entity in entity_list:
+        if is_valid_entity(entity, search_fields, keywords):
+            recall_set.append(entity)
+    return recall_set
+
+def is_valid_entity(entity, search_fields, keywords):
+    field_values = []
+    for search_field in search_fields:
+        if search_field in entity:
+            field_values.append(entity[search_field])
+    field_val_concat = ' '.join(field_values).lower()
+    is_valid = True
+    for keyword in keywords.split(' '):
+        try:
+            if keyword.lower() not in field_val_concat:
+                is_valid = False
+                break
+        except Exception as ex:
```

Comment:
It will be great to not catch any exceptions, but only a handful of expected errors.